stream.go

```diff
@@ -155,6 +160,11 @@ func (cs *clientStream) Trailer() metadata.MD {
 }
 
 func (cs *clientStream) SendMsg(m interface{}) (err error) {
+	cs.mu.Lock()
```

Comment:
I thought this approach was better since it limited the number of places touching the global `EnableTracing` var.

Keep in mind that this is a per-stream lock, not a global lock, so it only is contended when the stream itself is busy. That's not often the case (since streams usually require a specific ordering), so the mutex overhead will normally be unnoticed. I can run the benchmarks if you'd like me to verify this.